end-to-end-tests/cypress/support/commands.js

```diff
@@ -39,37 +43,48 @@ Cypress.Commands.add("navToEditorByDrawer", (viewerLinkText, navigationProfile)
   }
   cy.visit(Cypress.env("base_url"));
 
-  // Wait 5s for the events to start firing. In actuality it probably doesn't need to be
-  // this log but it also gives time for the handler above to fire as well.
-  cy.wait(5000);
-
   // Wait for any requests that were made to finish. We give all requests 1m. They
   // probably don't need that long, but they can have it.
   if (requestsMade.length) {
     cy.wait(requestsMade, { timeout: 60000 });
   }
+  
+  cy.get('ul[aria-label="file system navigator"]')
+    .find('li', { timeout: 15000 })
+    .should('have.attr', 'aria-expanded', 'false') 
+    .contains(oscalType)
+    .click();
 
-  cy.get("p")
-    .contains(viewerLinkText)
-    .should("be.visible")
-    .click({ timeout: 100000 });
-  cy.contains(navigationProfile, { timeout: 10000 }).click();
-});
+  cy.contains(pageTitle).click();
 
-Cypress.Commands.add("navToSspEditor", (toNavigate) => {
-  cy.navToEditorByDrawer("System Security Plan", toNavigate);
-});
-
-Cypress.Commands.add("navToCdefEditor", (toNavigate) => {
-  cy.navToEditorByDrawer("Component", toNavigate);
+  if (shouldWait) {
+    cy.waitForLoad();
```

Comment:
Why don't we just always do this

Reply:
I considered that, but I thought that navigating to a page does not strictly require waiting for data to load. Maybe I am getting to caught up on the naming of the function. I can change that functionality if you feel like its best.

Reply:
So I just have a huge ick for functions that have a boolean argument that toggles behavior. Because when we call this, we have no idea what the `true` or `false` does.

```ts
    cy.navToCdefEditor(MONGODB_NAVIGATION);
    cy.navToSspEditor(SSP_NAVIGATION);
    cy.navToProfileEditor(PROFILE_NAVIGATION_V4);
    cy.navToCatalogEditor(CATALOG_NAVIGATION);
    cy.navToCdefEditor(MONGODB_NAVIGATION);
    cy.navToCdefEditor(COMPONENT_NAVIGATION);
    cy.navToProfileEditor(PROFILE_NAVIGATION_V5);
    cy.navToProfileEditor(PROFILE_NAVIGATION_V5, true);
```

What does `true` mean for the last one? Why do none of the others have it? Does `undefined` mean something different from `false`? None of that is actually explicitly documented nor implied by a random positional boolean argument.

I would strongly prefer that we either just always call `waitForLoad`, that we call it when needed at the top level, or that create a separate function `navToAndWaitForFoo` (but that's wordy and leads to an explosion of command names).

So:

```ts
    cy.navToCdefEditor(MONGODB_NAVIGATION);
    cy.navToSspEditor(SSP_NAVIGATION);
    cy.navToProfileEditor(PROFILE_NAVIGATION_V4);
    cy.navToCatalogEditor(CATALOG_NAVIGATION);
    cy.navToCdefEditor(MONGODB_NAVIGATION);
    cy.navToCdefEditor(COMPONENT_NAVIGATION);
    cy.navToProfileEditor(PROFILE_NAVIGATION_V5);
    cy.navToProfileEditor(PROFILE_NAVIGATION_V5, true);
    cy.waitForLoad();
```

or:

```ts
    cy.navToCdefEditor(MONGODB_NAVIGATION);
    cy.navToSspEditor(SSP_NAVIGATION);
    cy.navToProfileEditor(PROFILE_NAVIGATION_V4);
    cy.navToCatalogEditor(CATALOG_NAVIGATION);
    cy.navToCdefEditor(MONGODB_NAVIGATION);
    cy.navToCdefEditor(COMPONENT_NAVIGATION);
    cy.navToProfileEditor(PROFILE_NAVIGATION_V5);
    cy.navToAndWaitForProfileEditor(PROFILE_NAVIGATION_V5);
```

Reply:
I went for needing to explicitly need to wait for the data to load. I still think that navigating to a page should not always require waiting, especially given your example above. Needing to wait for all data to appear there would make that test take a ton of time, when I do not think it is necessary to do so.